react-native-git-upgrade/cli.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+require('babel-register')({
+  presets: [
+    require('babel-preset-es2015-node'),
+    require('babel-preset-stage-3')
```

Comment:
My trouble with promise chains is that they look easy on paper:
```
A.then(B).then(C)
```
Unfortunately they end up really complicated when it comes to error handling and recovery from errorrs
```
A.then(B, ()=> {onError.then(B)}.then(C, onError())
```